static/app/utils/profiling/profile/sampledProfile.tsx

```diff
@@ -31,18 +31,44 @@ export class SampledProfile extends Profile {
 
     for (let i = 0; i < sampledProfile.samples.length; i++) {
       const stack = sampledProfile.samples[i];
-      const weight = sampledProfile.weights[i];
+      let weight = sampledProfile.weights[i];
 
-      profile.appendSampleWithWeight(
-        stack.map(n => {
-          if (!frameIndex[n]) {
-            throw new Error(`Could not resolve frame ${n} in frame index`);
-          }
+      let resolvedStack = stack.map(n => {
+        if (!frameIndex[n]) {
+          throw new Error(`Could not resolve frame ${n} in frame index`);
+        }
 
-          return frameIndex[n];
-        }),
-        weight
-      );
+        return frameIndex[n];
+      });
+
+      if (
+        i > 0 &&
+        resolvedStack[resolvedStack.length - 1]?.name ===
```

Comment:
Do we know how deep this stack is? From what I've seen, it's always 2 frames deep. If so, can we add the extra condition? This way, when this is implemented on the backend, it doesn't break anything

Reply:
Ok, made the change + left a note for our future selves in the comment 📦